src/main/java/analysis_use_case/Analysis.java

```diff
@@ -14,62 +18,73 @@ public Analysis(Inventory inventory){
     }
 
     /**
-     * Prints a revenue breakdown line by line
-     * Each line should contain the item's name, quantity sold, selling price, total revenue for that product
-     * Check if each item is quantified by weight, if so print the information accordingly
-     * Print out the total revenue of the inventory
+     * Return a revenue breakdown
+     * Each index should contain the item's name, quantity sold, selling price, total revenue for that product
+     * Check if each item is quantified by weight, if so present the information accordingly
+     * The last index should have the total revenue of the inventory
      */
-    public void totalRevenueBreakdown(){
-        System.out.println("Revenue:");
-        for (InventoryItem item : inventory.getItems()){
-            if (item.getIsWeight()){
-                System.out.println(item.getName() + ": kg sold: " + item.getQuantitySold() +
-                                   " price per kg: " + item.getSellPrice() + " total revenue for product: " +
-                                   item.getQuantitySold() * item.getSellPrice());
+    public ArrayList<String> totalRevenueBreakdown() {
```

Comment:
Instead of printing the information (which was violating clean architecture) I store the information in a ArrayList of Strings where each index is a breakdown of the item's revenue. This made it when I had to create the UI.